### PR TITLE
Contracts/feature/new specs

### PR DESCRIPTION
**Pull request by hacker-DOM**
Implemented everything I wanted to implement.
I'll have a look at Alex's solution for #96, other than that, we're done!

### CODE REVIEW

**Comment by dteiml** [comment]
guys

**Comment by josojo** [comment]
approved!
Thanks for breaking the commits into clear steps. Great! Love it

File: contracts/DutchExchange.sol [hunk 1]
```diff
@@ -583,9 +583,9 @@ contract DutchExchange {
         if (approvedTokens[buyToken] == true && approvedTokens[sellToken] == true) {
             // Get tulips issued based on ETH price of returned tokens
             if (buyToken == ETH) {
-                tulipsToIssue = buyerBalance * price.num / price.den;
-            } else if (sellToken == ETH) {
                 tulipsToIssue = buyerBalance;
+            } else if (sellToken == ETH) {
+                tulipsToIssue = buyerBalance * price.num / price.den;
```
Comment (W3stside) [inline contracts/DutchExchange.sol]: ~~should this not be `tulipsToIssue = buyerBalance * price.den / price.num;` as per line `580`~~

Comment (W3stside) [reply contracts/DutchExchange.sol]: nope scratch that, it's good

**Pull request closed without merging**